### ISSUE TITLE
[Feature]: Snooty able to be installed into isolated venv with sufficient Python version

### ISSUE THREAD

**Issue by TBBle** [vscode-restructuredtext/vscode-restructuredtext#344]
### Is your feature request related to a problem? Please describe.

I work on several projects using older Python versions, e.g. Python 2.7, and also on projects which manage their own venvs using tools like Poetry. And in one project, we're in a monorepo where I am switching venvs while working as I move around the project, and the (shared) Sphinx doc system is separate from _all_ of those projects.

In all of those cases, I don't want or can't have Snooty in the active Python env using for evaluating my code, i.e. even if the venv is has recent-enough Python, I'd hit #333.

I also don't want random things added to my system/user install, as requested in #258, I use pipx and Poetry to ensure isolation from user-local configuration like that.

The prompt for me lodging this here is that every time I start VS Code, even in projects that don't have any RST docs, I get a complained from Snooty about the language server, as I _basically never_ have VSCode running outside a venv for Python 3.

### Describe the solution you'd like

I'd like Snooty to be installed by the extension in a venv owned and managed by the extension, using a sufficiently-new version of Python, irrespective of the version of Python used by the project configuration.

Finding a sufficiently-new version of Python shouldn't be _too_ hard, on as Linux/MacOS the binary should be in the path as `python310`, `python39`, `python38`, or `python37`; and on Windows you can similarly use `py -3.10`, `py -3.9`, `py -3.8`, or `py -3.7`, until one is found that works.

I don't know if Snooty depends on having other things installed in the venv with it, e.g. Sphinx and extensions, but if so, I guess it'd make sense to be able to point to a requirements.txt or similar to add to the venv.

This should be an optional mode, I guess, since both #258 exists for installing Snooty into the system/user Python environment directly (for some value of "global interpreter"), and I imagine people with their Sphinx setup in their project venv would also want their project's venv to be used for Snooty too.

### Describe alternatives you've considered

I've considered #258, but I think it goes against my preference for a clean/minimal global install.

If #333 is fixed, then using the active venv would be an option, but in some cases that's impossible (Python 2 work), and in some cases it's undesirable (installing stuff in a managed venv, that will be uninstalled by other tools, and having to do that for every venv used by the monorepo), and those two cases cover the majority of my Python development work.

Another option would be to be able to select a venv or global interpreter for Snooty, like the Python extension offers, but independent of the interpreter active for the Python extension. That would mean for me, I'd need to manually create the venv somewhere and then point the extension at it to install Snooty and anything else it needs, and repeat that on all my dev machines. But it wouldn't need to be repeated per-project.

A final option that comes to mind would be to have the extension support pointing at an external Snooty install, as you can do with the Black code formatter, and then I could use pipx to install Snooty in a venv of its own, and then (I assume) there'd be a snooty.exe in the path. (This assumes Snooty can be used that way.) This one is basically the previous one, but a bit simpler for the user. Given the previous solution, I could use `pipx` to install Snooty, and then point the extension at the resulting venv, to achieve this solution, except that the path may vary across machines, while `snooty.exe` being called from the `PATH` means I don't need to specify per-host settings.

These last two options are a bit more manual than I'd prefer, but being _able_ to do that last one would allow working around any other Snooty installation issues by using a self-provided Snooty install through any means I like, e.g., `pipx` or `pip install --user` both should end up with `snooty.exe` in the `PATH`.

**Comment by lextm** [comment]
Switching to esbonio seems to have resolved this partially, as esbonio works with venv better (at least one user reported back).

You might try the new releases, such as 176.0.0 and let me know if any critical issue remains.

**Comment by lextm** [comment]
After another round of investigation, I think this issue will be closed now as won't fix.

The main reasons you raised this issue are,

1. Snooty has a requirement on Python >=3.7
2. Some of your projects are in virtual environments with older Python versions (2.7 to 3.6).

Why I think this issue should be closed as won't fix?

* Esbonio lowers the bar to Python >=3.6
* Python <3.7 are end of life, https://www.python.org/downloads/

Any effort on supporting end of life releases prevents the actual priorities (such as improving esbonio language server in every possible ways), and doesn't serve the general user base well.

Pull requests are welcome if anyone steps up to address it.

**Comment by TBBle** [comment]
The other reason I opened this issue is that I wanted this to _not_ be installed into a venv where I am doing other things, or into a non-venv install. That's "isolated" in the title, and everything but the first sentence of the description.

https://github.com/vscode-restructuredtext/vscode-restructuredtext/issues/258#issuecomment-1063878873 is roughly the same use-case, #258 wanted it installed in the user's site-package, I want it isolated so that:
* Other things I am doing don't break esbonio
* esbonio doesn't break other things I am working on
* _Nothing_ is installing packages into the global or user site-packages directory.

I've noticed while cleaning up old Python installs that the vscode Python extension is (or was) also installing random stuff into my user site-packages directory rather than maintaining its own venv; I suspect this is because it expects to be able to import stuff from my project (although none of my projects use the user site-packages, it contains only `pip`, `setuptools`, and `wheel` as I use venvs for _everything else_), but it's still pretty messy, and doesn't work well with multi-project monorepos (which is actually my main work-related Python usecase now).
